sdk/python/lib/pulumi/invoke.py

```diff
@@ -131,7 +131,9 @@ def merge(
             if source.plugin_download_url is None
             else source.plugin_download_url
         )
-        dest.version = dest.version if source.version is None else source.version
+        dest.version = (
+            dest.version if (source.version in [None, ""]) else source.version
```

Comment:
Maybe we could change the version to default to None instead, but that is a more invasive change than this.

Reply:
Changing to None does feel more correct, I'd imagine given it's typed as Optional[str] most places should be handling the None case already.